core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -133,72 +133,67 @@ class SYCLTeamMember {
     const unsigned int team_rank_ = team_rank();
 
     // First combine the values in the same subgroup
+#if defined(KOKKOS_ARCH_INTEL_GPU) || defined(KOKKOS_IMPL_ARCH_NVIDIA_GPU)
+    auto shuffle_combine = [&](int shift) {
+      if (vector_range * shift < sub_group_range) {
+        const value_type tmp = sg.shuffle_down(value, vector_range * shift);
+        if (team_rank_ + shift < team_size_) reducer.join(value, tmp);
+      }
+    };
+    shuffle_combine(1);
+    shuffle_combine(2);
+    shuffle_combine(4);
+    shuffle_combine(8);
+    shuffle_combine(16);
```

Comment:
Any comment (here on GH) about the dropping `shuffle_combine(32);`.  How does it impact the performance measurement you did?

Reply:
I don't see any significant impact in terms of performance. If the subgroup size is at most 32, we will just never use shuffle memory between threads ids that differ by 32.

Reply:
Right.  Just wanted to make sure there is some discussion of it and it is not just a silent last minute removal.